pallets/pools/src/tranche.rs

```diff
@@ -205,10 +592,10 @@ where
 	}
 
 	pub fn calculate_weights(&self) -> Vec<(Weight, Weight)> {
-		let n_tranches: u32 = self.0.len().try_into().expect("MaxTranches is u32");
+		let n_tranches: u32 = self.tranches.len().try_into().expect("MaxTranches is u32");
 		let redeem_starts = 10u128.checked_pow(n_tranches).unwrap_or(u128::MAX);
 
-		self.0
+		self.tranches
```

Comment:
We should add a comment explaining this logic. Something like

```suggestion
		// The desired order priority is:
		// - Senior redemptions
		// - Junior redemptions
		// - Junior investments
		// - Senior investments
		// We ensure this by having a higher base weight for redemptions,
		// increasing the redemption weights by seniority,
		// and decreasing the investment weight by seniority.
		self.tranches
```

Reply:
Done.